sync3/caches/user.go

```diff
@@ -371,7 +372,13 @@ func (c *UserCache) LazyLoadTimelines(ctx context.Context, loadPos int64, roomID
 		}
 
 		result[roomID] = urd
-		c.roomToData[roomID] = urd
+		// only replace our knowledge if it is the future
+		// TODO FIXME: this is an incomplete solution as all the non-Timeline fields we return in this
+		// function will be AHEAD of the load pos provided, meaning you could see a stale timeline with
+		// incorrect notif counts (until you've consumed the channel)
```

Comment:
As discussed: this is unfortunate, but:
- it will self-heal shortly as the channel of updates is consumed
- it is a less-bad failure mode than missing messages!